crates/volta-core/src/tool/yarn/fetch.rs

```diff
@@ -122,7 +126,19 @@ fn determine_remote_url(version: &Version, hooks: Option<&ToolHooks<Yarn>>) -> F
             let distro_file_name = Yarn::archive_filename(&version_str);
             hook.resolve(version, &distro_file_name)
         }
-        _ => Ok(public_registry_package("yarn", &version_str)),
+        _ => {
+            let matches_yarn_berry = VersionReq::parse(">=2").unwrap();
```

Comment:
Since we're only looking for a major version difference and we have a `Version` object, I think we can check the [`major`](https://docs.rs/semver/latest/semver/struct.Version.html#structfield.major) field of the `Version` directly, rather than creating a `VersionReq` and verifying if it matches. Something like `version.major >= 2`

Reply:
🤦  I didn't realize we could just check the major field directly. That's so much nicer! Fixed in db9e6d1d0e1a2ca910231af66ce65b8f9416f382